orchestration/dagster_orchestration/hca_orchestration/resources/beam.py

```diff
@@ -77,67 +99,102 @@ def dispatch_k8s_job(namespace: str, image_name: str, job_name_prefix: str, args
     "subnet_name": Field(StringSource),
+    "region": Field(StringSource),
+    "worker_machine_type": Field(StringSource),
+    "starting_workers": Field(IntSource),
+    "max_workers": Field(IntSource),
     "service_account": Field(StringSource),
+    "kubernetes_service_account": Field(StringSource),
+    "temp_bucket": Field(StringSource),
     "image_name": Field(StringSource),
     "image_version": Field(StringSource),
-    "namespace": Field(StringSource)
+    "namespace": Field(StringSource),
 })
-def dataflow_beam_runner(init_context: InitResourceContext):
-    return DataflowBeamRunner(
+def base_dataflow_beam_runner(init_context: InitResourceContext):
+    cloud_config = DataflowCloudConfig(
         project=init_context.resource_config['project'],
-        temp_location=init_context.resource_config['temp_location'],
-        subnet_name=init_context.resource_config['subnet_name'],
         service_account=init_context.resource_config['service_account'],
+        subnet_name=init_context.resource_config['subnet_name'],
+        region=init_context.resource_config['region'],
+        worker_machine_type=init_context.resource_config['worker_machine_type'],
+        starting_workers=init_context.resource_config['starting_workers'],
+        max_workers=init_context.resource_config['max_workers'],
+    )
+    return DataflowBeamRunner(
+        cloud_config=cloud_config,
+        kubernetes_service_account=init_context.resource_config['kubernetes_service_account'],
+        temp_bucket=init_context.resource_config['temp_bucket'],
         image_name=init_context.resource_config['image_name'],
         image_version=init_context.resource_config['image_version'],
-        namespace=init_context.resource_config['namespace']
+        namespace=init_context.resource_config['namespace'],
+        logger=init_context.log_manager,
     )
 
 
+@configured(base_dataflow_beam_runner)
+def dataflow_beam_runner(config):
+    return {
+        'temp_bucket': {'env': 'HCA_TEMP_STORAGE_BUCKET'},
+        'image_name': {'env': 'TRANSFORM_PIPELINE_IMAGE'},
+        'image_version': {'env': 'TRANSFORM_PIPELINE_IMAGE_VERSION'},
+        'service_account': {'env': 'HCA_DATAFLOW_SERVICE_ACCOUNT'},
+        'kubernetes_service_account': {'env': 'HCA_KUBERNETES_SERVICE_ACCOUNT'},
+        'project': {'env': 'HCA_GOOGLE_PROJECT'},
+        'subnet_name': {'env': 'DATAFLOW_SUBNET_NAME'},
+        'region': {'env': 'GCLOUD_REGION'},
+        'worker_machine_type': {'env': 'DATAFLOW_WORKER_MACHINE_TYPE'},
+        'starting_workers': {'env': 'DATAFLOW_STARTING_WORKERS'},
+        'max_workers': {'env': 'DATAFLOW_MAX_WORKERS'},
+        'namespace': {'env': 'KUBERNETES_NAMESPACE'},
+        **config,
+    }
+
+
 @dataclass
 class LocalBeamRunner:
     working_dir: str
+    logger: DagsterLogManager
+    # TODO this is hardcoded to the HCA transformation pipeline for now
+    target_class: str = 'hca-transformation-pipeline'
```

Comment:
❓ worth a small bite ticket to parameterize this?

Reply:
i can definitely do it as part of this, it seems even smaller than a small bite imo